src/Aggregate/ConfigurableAggregateTranslator.php

```diff
@@ -27,92 +26,55 @@ class ConfigurableAggregateTranslator implements AggregateTranslator
     /**
      * @var string
      */
-    private $versionMethodName = 'getVersion';
+    private $versionMethodName;
 
     /**
      * @var string
      */
-    private $identifierMethodName = 'getId';
+    private $identifierMethodName;
 
     /**
      * @var string
      */
-    private $popRecordedEventsMethodName = 'popRecordedEvents';
+    private $popRecordedEventsMethodName;
 
     /**
      * @var string
      */
-    private $replayEventsMethodName = 'replay';
+    private $replayEventsMethodName;
 
     /**
      * @var string
      */
-    private $staticReconstituteFromHistoryMethodName = 'reconstituteFromHistory';
+    private $staticReconstituteFromHistoryMethodName;
 
     /**
      * @var null|callable
      */
-    private $eventToMessageCallback = null;
+    private $eventToMessageCallback;
 
     /**
      * @var null|callable
      */
-    private $messageToEventCallback = null;
+    private $messageToEventCallback;
 
     /**
-     * @param null|string   $identifierMethodName
-     * @param null|string   $versionMethodName
-     * @param null|string   $popRecordedEventsMethodName
-     * @param null|string   $replayEventsMethodsName
-     * @param null|string   $staticReconstituteFromHistoryMethodName
-     * @param null|callable $eventToMessageCallback
-     * @param null|callable $messageToEventCallback
+     * @param null|AggregateTranslatorConfiguration $configuration
      */
-    public function __construct(
-        $identifierMethodName = null,
-        $versionMethodName = null,
-        $popRecordedEventsMethodName = null,
-        $replayEventsMethodsName = null,
-        $staticReconstituteFromHistoryMethodName = null,
-        $eventToMessageCallback = null,
-        $messageToEventCallback = null)
+    public function __construct(AggregateTranslatorConfiguration $configuration = null)
```

Comment:
Please have a look again. I admit it's kind of back and forth using the config without having a BC break but the intention is to have the config as single point of validation for itself.
Thanks for the hint @codeliner I changed the config's constructor to private.